lib/wallet/src/Cardano/Wallet/Write/Tx/Balance/TokenBundleSize.hs

```diff
@@ -52,38 +51,24 @@ instance NFData TokenBundleMaxSize
 --
 -- See 'W.TokenBundleSizeAssessor' for the expected properties of this function.
 --
-tokenBundleSizeAssessor :: TokenBundleMaxSize -> TokenBundleSizeAssessor
-tokenBundleSizeAssessor maxSize =
-    TokenBundleSizeAssessor { assessTokenBundleSize }
+tokenBundleSizeAssessor
+    :: RecentEra era
+    -> PParams (ShelleyLedgerEra era)
+    -> TokenBundleSizeAssessor
+tokenBundleSizeAssessor era pp = TokenBundleSizeAssessor $ \tb ->
+    let
+    in
+        if serSize (toLedger tb) > maxValSize
+        then TokenBundleSizeExceedsLimit
+        else TokenBundleSizeWithinLimit
   where
-    assessTokenBundleSize tb
-        | serializedLengthBytes <= maxSize' =
-            TokenBundleSizeWithinLimit
-        | otherwise =
-            TokenBundleSizeExceedsLimit
-      where
-        serializedLengthBytes :: TxSize
-        serializedLengthBytes = computeTokenBundleSerializedLengthBytes tb
-
-        maxSize' :: TxSize
-        maxSize' = unTokenBundleMaxSize maxSize
+    maxValSize :: TxSize
+    maxValSize = TxSize $ withConstraints era $ pp ^. ppMaxValSizeL
 
-computeTokenBundleSerializedLengthBytes :: TokenBundle.TokenBundle -> TxSize
-computeTokenBundleSerializedLengthBytes =
-    TxSize
-        . safeCast
-        . BS.length
-        . serialize' shelleyProtVer
-        . Cardano.toMaryValue
-        . toCardanoValue
-  where
-    safeCast :: Int -> Natural
-    safeCast = fromIntegral
+    protVer :: ProtVer
+    protVer = withConstraints era $ pp ^. ppProtocolVersionL
 
--- | Get a 'TokenBundleMaxSize' from a 'PParam era'.
-getTokenBundleMaxSize
-    :: RecentEra era
-    -> PParams (ShelleyLedgerEra era)
-    -> TokenBundleMaxSize
-getTokenBundleMaxSize era pp = withConstraints era $
-    TokenBundleMaxSize $ TxSize $ pp ^. ppMaxValSizeL
+    serSize :: Value StandardCrypto -> TxSize
+    serSize = fromIntegral @Int64 @TxSize
```

Comment:
Perhaps this should fail instead of silently wrapping and giving the wrong answer.

Reply:
Done